app/models/manageiq/providers/autosde/storage_manager/cloud_volume.rb

```diff
@@ -10,12 +10,24 @@ class ManageIQ::Providers::Autosde::StorageManager::CloudVolume < ::CloudVolume
 
   def self.raw_create_volume(ext_management_system, options = {})
     # @type [StorageService]
-    vol_to_create = ext_management_system.autosde_client.VolumeCreate(
-      :service => ext_management_system.storage_services.find(options["storage_service_id"]).ems_ref,
+    creation_hash = {
+      :service => "",
       :name    => options["name"],
       :size    => options["size"],
       :count   => options["count"]
-    )
+    }
+
+    if options['mode'] == 'Basic'
+      creation_hash[:service] = ext_management_system.storage_services.find(options["storage_service_id"]).ems_ref
+    else
+      creation_hash[:service_name] = options["new_service_name"]
+      creation_hash[:resources] = []
+      options["storage_resource_id"].each do |resource|
+        creation_hash[:resources].push(ext_management_system.storage_resources.find(resource['value']).ems_ref)
+      end
```

Comment:
A more efficient way of doing this would be
```suggestion
      creation_hash[:resources] = ext_management_system.storage_resources.find(options["storage_resource_id"].to_a).pluck(:ems_ref)
```